src/opossum_lib/opossum_file.py

```diff
@@ -98,6 +98,25 @@ def element_exists_but_resource_type_differs(
             return self.children[element].type != resource_type
         return False
 
+    def drop_element(
```

Comment:
Right now the `add_path` mutates the `Resource` while `drop_element` returns a mutated copy. Maybe would be easier to be consistent and decided for one of the two ways.
Actually, as the `Resource` class is declared as `frozen`, it is a bad practice to mutate the `children` dict (as it is done in the `add_path` function).
So I would recommend that we either remove the `frozen=true` from the class and change the `drop_element` to mutate the existing instance or we change the `add_path` to return a mutated copy.

Reply:
I decided to keep the `frozen` property and changed the `add_path` method accordingly.